frontends/concrete-python/concrete/fhe/mlir/utils.py

```diff
@@ -187,8 +190,22 @@ def construct_deduplicated_tables(
                 [ [5, 8, 6, 7][input[2, 0]] , [3, 1, 2, 4][input[2, 1]] ]
     """
 
+    raw_table = construct_table(node, preds)
+    if all(isinstance(value, int) for value in raw_table):
+        return ((np.array(raw_table), None),)
+
     node_complete_table = np.concatenate(
-        tuple(np.expand_dims(array, -1) for array in construct_table(node, preds)),
+        tuple(
+            np.expand_dims(
+                (
+                    array
+                    if isinstance(array, np.ndarray)
+                    else np.broadcast_to(array, node.output.shape)
+                ),
+                -1,
+            )
+            for array in raw_table
+        ),
         axis=-1,
```

Comment:
@umut-sahin where is the test please? we should always have tests when we fix a problem

Reply:
This is tested by every table lookup in the test suite :slightly_smiling_face:

Reply:
We can't easily test the memory usage in a 64 thread test suite.

Reply:
Also, we have %100 test coverage, CI won't go green without entering this branch.

Reply:
the test was there but did not catch the problem, so I think the test should have been extended or another one should have been added, to catch the problem. Like, just trying to compile an MLIR like the one I added in the issue, and see if it compiles

Reply:
coverage: yes, but still, I think every issue should come with tests to avoid they are back, every issue

Reply:
created https://github.com/zama-ai/concrete/pull/587 for it

We couldn't test it before as it'd crash the test suite. We could have realized the issue is there, but the issue will always be there, if you use `(3000, 1000, 2000)` for example, it'll crash...

We just didn't know the right limit.

Reply:
thx. In my view, the new test of #587 would have been added to this #584 issue. The fix at the same time than the new test